src/CodeGen_C.cpp

```diff
@@ -315,7 +315,7 @@ class TypeInfoGatherer : public IRGraphVisitor {
     std::set<Type> vector_types_used;
 };
 
-std::string maybe_emit_numeric_constant(const Expr &e) {
+std::string maybe_print_numeric_constant(const Expr &e) {
```

Comment:
I would call this "stringify_if_constant" or "text_if_constant". "Maybe" is nice if one wants to give the routine a sense of mystery... One line comment "// Returns text of value if constant, empty string otherwise" would also be helpful. (Yes, it's a total nitpick comment at the top of a code review. My apologies, I find the "maybe" pattern questionable, but maybe it's just that I'm allergic to that Carly Rae Jepsen song :-))

Reply:
Maybe a bit more constructively, what happens if we move the 16-bit float handling logic here into the visitor for ```FloatImm```?

Reply:
> Maybe a bit more constructively, what happens if we move the 16-bit float handling logic here into the visitor for FloatImm?

This case doesn't use that visitor for emitting anything, so, we'd need to restructure code to ensure that visitation happened when emitting all FloatImm values.